app/scripts/directives/lifecycleHook.js

```diff
@@ -130,4 +130,15 @@ angular.module("openshiftConsole")
         });
       }
     };
+  })
+  .directive("lifecycleHook", function() {
+    return {
+      restrict: 'E',
+      scope: {
+        hookParams: "=",
+        namespace: "=",
+        type: "@"
```

Comment:
I think this is better as

```js
scope: {
  deploymentConfig: '=',
  type: '@'  // "pre", "mid", or "post"
}
```

Should be able to determine strategy from the deployment config spec, correct? Then you don't need to pass namespace because it's in the deployment config metadata.